lib/imap.php

```diff
@@ -97,9 +97,10 @@ public function checkPassword($uid, $password) {
 		curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
 		curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'CAPABILITY');
 
-		$canconnect = curl_exec($ch);
+		curl_exec($ch);
+		$errorcode = curl_errno($ch);
 
-		if($canconnect) {
+		if($errorcode == 0) {
```

Comment:
could you use an explicit comparison  (`===`)?

Reply:
FYI I've manually used this fix in one of my nextcloud 20.0.10 instance, and it has resolved the issue I was facing with IMAP users having 0 email in their mailbox not being able to login, as wrong username/password message was coming back. THX! This shall go into next release of nextcloud.

`               curl_exec($ch);
                $errorcode = curl_errno($ch);

                if($errorcode === 0) {`

Reply:
I made the requested updates to my code.